src/lib/graft/task.cpp

```diff
@@ -973,9 +975,14 @@ void PeriodicTask::finalize()
 
 std::chrono::milliseconds PeriodicTask::getTimeout()
 {
-    auto ret = (m_initial_run) ? m_initial_timeout_ms : m_timeout_ms;
-    m_initial_run = false;
-    return ret;
+    if(m_initial_run)
+    {
+        m_initial_run = false;
+        return m_initial_timeout_ms;
+    }
+    if(m_random_factor < 0.0001) return m_timeout_ms;
+
+    return std::chrono::milliseconds( int(m_timeout_ms.count() * (1.0 + std::rand()*m_random_factor/RAND_MAX) ));
```

Comment:
`std::rand()` is depreciated in modern C++, and it's also not random at all if you don't have an `std::srand()` call somewhere in the program.  Far better to replace this with more modern C++:

```C++
#include <random>

// ...

    static std::mt19937 rng{std::random_device{}()};
    int secs = m_timeout_ms.count();
    return std::chrono::milliseconds{
        std::uniform_int_distribution<int>{secs, int(secs*m_random_factor)}(rng)};
```

Reply:
Thank you. I have done, but for the purpose it was good enough.